### ISSUE TITLE
cmpZero does not terminate

### ISSUE THREAD

**Issue by AlexandreTunstall** [konn/type-natural#23]
Expected

```
> cmpZero sOne
Refl
> cmpZero Zero
Refl
```

but got

```
> cmpZero sOne
*** Exception: stack overflow
> cmpZero Zero

```

The 0 case never seems to reach a stack overflow, it just loops forever with constant memory usage. Nonetheless, the function is currently unusable.

Functions in the `Data.Type.Natural.Lemma.Order` module that don't depend on `cmpZero` seem to work fine.

In case it's version-specific:

* GHC 8.10.7
* type-natural 1.1.0.0
* constraints 0.13
* equational-reasoning 0.7
* ghc-typelits-knownnat 0.7.6
* ghc-typelits-natnormalise 0.7.6
* ghc-typelits-presburger 0.6.0.0
* integer-logarithms 1.0.3.1

**Comment by AlexandreTunstall** [comment]
It might be caused by the following cycles in the function's definition.

In the 0 case:

```
sCmpNat sZero
  => leqToLT sZero (sSucc sZero) _
  => ltSucc sZero
  => cmpZero sZero
```

In every other case:

```
sCmpNat (sSucc n)
  => leqToLT sZero (sSucc (sSucc n)) _
  => ltSuccLToLT sZero (sSucc (sSucc n)) _
  => cmpSuccStepR sZero (sSucc n)
  => cmpZero sZero
```

**Comment by konn** [comment]
Thank you for pointing it out, and sorry for my late response.
As it can be solved automatically by type-checker plugin, I will release the fixed version as soon as possible.

**Comment by konn** [comment]
I've just released type-natural 1.1.0.1, which removes infinite loops from order-theoretic lemmas. Thank you again for reporting!
